projects/observability/src/shared/dashboard/data/graphql/table/entity/entity-table-data-source.model.ts

```diff
@@ -117,15 +105,26 @@ export class EntityTableDataSourceModel extends TableDataSourceModel {
 
   private queryEntityChildren(
     entity: Entity,
-    request: TableDataRequest<SpecificationBackedTableColumnDef>
+    parentRequest: TableDataRequest<SpecificationBackedTableColumnDef>
   ): Observable<TableRow[]> {
-    return this.query<EntitiesGraphQlQueryHandlerService>(this.buildChildEntityRequest(request, entity)).pipe(
-      map(response => this.resultsAsTreeRows(response.results, request, false))
+    this.childEntityDataSource!.parentEntityFilter = GraphQlEntityFilter.forEntity(entity);
+
+    return this.childEntityDataSource!.getData().pipe(
+      mergeMap(child => child.getData(this.buildChildTableRequest(parentRequest))),
+      map(response => this.resultsAsTreeRows(response.data as Entity[], parentRequest, false))
     );
   }
-}
 
-interface EntityTableGraphQlRequest extends GraphQlEntitiesQueryRequest {
-  tableRequestType: 'root' | 'children';
-  tableRequest: TableDataRequest;
+  private buildChildTableRequest(
+    parentRequest: TableDataRequest<SpecificationBackedTableColumnDef>
+  ): TableDataRequest<SpecificationBackedTableColumnDef> {
+    return {
+      columns: parentRequest.columns,
+      position: {
+        startIndex: 0,
+        limit: 1000
```

Comment:
can we bump this limit down to 40 to match prior behavior? Or at least 100. no reason to fetch 1000 children if there happen to be that many

Reply:
Yes. Will update this.

Reply:
done